duckduckgo/iOS
# Harden content scripts

Task/Issue URL: https://app.asana.com/0/69071770703008/1200769105021917/f
Tech Design URL:
CC: @sammacbeth 

**Description**:
This PR hardens our content scripts against leaking global variables that can be exploited. In `contentblocker.js` we freeze the prototype of `Trackers` to prevent overriding properties or methods.
We also move `duckduckgoDebugMessaging` inside anonymous functions where it is used.

**Steps to test this PR**:
1. Go to a well known page with trackers and ensure blocked count is the same before and after change.
1. Go to [Privacy Test Pages](https://privacy-test-pages.glitch.me/security/js-leaks.html) and ensure `duckduckgoDebugMessaging` is no longer leaked.
2. Execute the following code in the console:
    1. `Trackers.prototype.getTrackerData = function() { console.log('Howdy doody'); }`
    2. `const track = new Trackers({ tldjs: {}, utils: {}})`
    3. `track2.getTrackerData() // Should produce an error: 'Error: tried to detect trackers before rules were loaded'`



**Device Testing**:


**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
